packages/rs-drive-abci/src/abci/handler/mod.rs

```diff
@@ -196,6 +199,13 @@ where
 
         let transaction = transaction_guard.as_ref().unwrap();
 
+        // Get current platform version
+        let state = self.platform.state.read().expect("expected to get state");
+        let current_protocol_version = state.current_protocol_version_in_consensus();
+        drop(state);
+        let platform_version = PlatformVersion::get(current_protocol_version)
+            .map_err(|e| proto::ResponseException::from(Error::from(e)))?;
```

Comment:
This is wrong, because you might be at a different platform version than the execution of the block proposal. Instead this should be checked on the block platform state, or the version can be added to the block execution outcome (run_result)